pennylane/tape/operation_recorder.py

```diff
@@ -45,15 +45,24 @@ class OperationRecorder(QuantumTape):
 
     def __init__(
         self, ops=None, measurements=None, prep=None, name=None, do_queue=False, _update=True
-    ):
-        super().__init__(
-            ops, measurements, prep=prep, name=name, do_queue=do_queue, _update=_update
-        )
+    ):  # pylint: disable=unused-argument
+        AnnotatedQueue.__init__(self)
+        QuantumScript.__init__(self, ops, measurements, prep, name=name, _update=_update)
         self.ops = None
         self.obs = None
 
-    def _process_queue(self):
-        super()._process_queue()
+    def __enter__(self):
+        return AnnotatedQueue.__enter__(self)
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        AnnotatedQueue.__exit__(self, exception_type, exception_value, traceback)
+        # After other optimizations in #2963, #2986 and follow-up work, we should check whether
+        # calling `_process_queue` only if there is no `exception_type` saves time. This would
+        # be done via the following:
+        # if exception_type is None:
+        #    self._process_queue()
+        self._ops, self._measurements, self._prep = process_queue(self)
+        self._update()

```

Comment:
Why is this behaviour different from `QuantumTape`? We are not using a lock nor queuing `OperationRecorder` when `self.do_queue = True`.

Reply:
tbh - I do not know what the lock is for, but the difference is that we're not queuing the OperationRecorder itself. This was a change from the Tape -> Script update, and I'm making that same change here. We're still not 100% sure if it even makes a difference, but also OperationRecorder was always different from QuantumTape in that it would manually append its queued operations to the surrounding context (if there was one), whereas QuantumTape would just queue _itself_ and let the surrounding context pop and expand the tape.

I can add back the lock if we're worried about it.

Reply:
I think this is just an oversight, but not 100% sure. The locking was added to the tape by an external contributor, as it fixed a race condition bug when used with multiprocessing. However, we might have just forgotten to add it here as well.

Reply:
I intentionally didn't add it here because I didn't think OperationRecorder was intended to be used like QuantumTape was. I can add it back

Reply:
Ah - in which case disregard my comment 🙃